packages/expo-dev-menu/ios/DevMenuRCTBridge.m

```diff
@@ -27,7 +27,7 @@ - (RCTDevMenu *)devMenu
 
 - (NSArray<Class> *)filterModuleList:(NSArray<Class> *)modules
 {
-  NSArray<NSString *> *allowedModules = @[@"RCT"];
+  NSArray<NSString *> *allowedModules = @[@"RCT", @"DevMenu"];
```

Comment:
This shouldn't be changed. Please, add `react-native-safe-area-context` to VendoredModules - https://github.com/expo/expo/blob/e1dab6a8a8a38fd4b2743d16d75baeca124f3b90/packages/expo-dev-menu/ios/DevMenuVendoredModulesUtils.m#L20-L31 instead and remove all `RCT_EXPORT_MODULE` macros from ios codebase.

Reply:
I tried to explain above that this method is invoked at a different time than the standard registry and leads to warnings - have you encountered this before?

Reply:
Ohh, sorry I thought you were referring to something else. Do you know exactly which part of the code causes that warning?

Reply:
no problem - the method is `constantsToExport` in `RNCSafeAreaProviderManager` which accesses the window frame which is understandably a UI thread thing

Reply:
if you add to the vendored modules and remove this filtered modules you should be able to reproduce - I can also double check later today

Reply:
No need. I think that `extraModules` can't be initialized on the main thread, but we can change this code and run it synchronously on the main thread to get rid of the warning. When we leave that macro, that would mean that this module will be registered in the app bridge too. That maybe isn't a big deal right now, but maybe in the future. So I would like to avoid that.

Reply:
makes sense - I've updated to wrap the method, also verified this is working on android :)